### DISCUSSION TITLE
Geolocation tracking for IOS apps using .net Maui

### DISCUSSION THREAD

**Discussion by ArmandKeulder** [dotnet/maui#19224] [Q&A]
I'm working on an in-house iOS app designed to help users accurately track their routes during trips. Currently, I've implemented a method to track users when the app is open in the background. However, I'm facing challenges, as the tracking stops when the device is locked for more than 10 minutes.

I'm looking for a solution to continuously track a user's geolocation, even if the app is closed or not in use. Specifically, I want to ensure uninterrupted tracking, especially when the device is locked.

Here are some key points:

Current Method: I'm currently using the Core Location method and a combination of background tasks and a repeating timer to fetch the user's location and update a log for geolocation tracking when the app is open in the background.

Issues Faced: The tracking stops when the device is locked for more than 10 minutes. This limitation impacts the accuracy of the route tracking during longer trips.

Objective: My goal is to achieve continuous geolocation tracking, even when the app is closed or not actively used, to provide users with a seamless and accurate record of their routes.

Platform: The app is developed for iOS using the .net maui platform, and I'm seeking solutions or suggestions that are compatible with the iOS .net maui environment.

If anyone has experience or insights into achieving continuous geolocation tracking on iOS, especially when the app is not in use or the device is locked, I would greatly appreciate the assistance.

**Comment by Xyncgas** [comment]
you have a bigger shot for talking to apple and be friends with them and let you do what you want to do because that's apple, you are on their device it's their rules

**Comment by spotwalla** [comment]
It's been a while since your post, but I'm in a similar position. All the information I've found indicate that CLLocationManager survives in the background and locked screens. And while I think I've found the right combination of configuration settings and get some degree of background updates, ultimately my app is still at Apple's discretion.

Have you found the magic sauce? Care to share?
